packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -20,16 +20,36 @@ const duotone = require(`./duotone`)
 const { IMAGE_PROCESSING_JOB_NAME } = require(`./gatsby-worker`)
 
 const imageSizeCache = new Map()
+
+const getImageSizeAsync = async file => {
+  if (
+    process.env.NODE_ENV !== `test` &&
+    imageSizeCache.has(file.internal.contentDigest)
+  ) {
+    return imageSizeCache.get(file.internal.contentDigest)
+  }
+  const input = fs.createReadStream(file)
+  const dimensions = await imageSize(input)
+
+  if (!dimensions) {
+    reportError(
+      `gatsby-plugin-sharp couldn't determine dimensions for file:\n${file.absolutePath}\nThis file is unusable and is most likely corrupt.`,
```

Comment:
This might be a bit too presumptuous as there could be other reasons why the dimension wasn't found, unrelated to the sanity of the image. I'd relax the force of the last sentence a bit.

Reply:
That's the wording from the current function, which I thought best to leave alone

Reply:
Oh, then I missed the move and I take it back. Out of scope.